app/src/main/java/bg/zahov/app/ui/settings/SettingsViewModel.kt

```diff
@@ -1,74 +1,101 @@
+ * @property auth Manages user authentication and provides user actions via [UserProviderImpl].
+ * @property workoutState Manages the current workout state, ensuring any ongoing workout is tracked and can be cancelled on user logout or account deletion.
+ *
+ * This ViewModel handles:
+ * - Retrieving and updating user settings.
+ * - Logging out the user and resetting their settings.
+ * - Deleting the user's account and associated data.
+ */
+class SettingsViewModel(
+    private val repo: SettingsProviderImpl = Inject.settingsProvider,
+    private val auth: UserProviderImpl = Inject.userProvider,
+    private val workoutState: WorkoutStateManager = Inject.workoutState
+) : ViewModel() {
 
-    private val repo by lazy {
-        application.getSettingsProvider()
-    }
-
-    private val auth by lazy {
-        application.getUserProvider()
-    }
-
-    private val workoutState by lazy {
-        application.getWorkoutStateManager()
-    }
+    /**
+     * Data class representing the state of the settings UI.
+     *
+     * @property data Current settings data.
+     * @property returnBack Indicates if the UI should navigate back to welcome screen(e.g., after logout or account deletion).
+     */
+    data class SettingsData(
+        val data: Settings,
+        val returnBack: Boolean = false
+    )
 
-    private val _state = MutableLiveData<State>()
-    val state: LiveData<State>
-        get() = _state
+    // Holds the current UI state, updated whenever settings data changes
+    private val _uiState = MutableStateFlow(SettingsData(data = Settings()))
+    val uiState: StateFlow<SettingsData> = _uiState
 
     init {
+        // Initializes settings by collecting data from repo
         viewModelScope.launch {
             repo.getSettings().collect {
-                it.obj?.let { settings -> _state.postValue(State.Data(settings)) }
+                it.obj?.let { settings ->
+                    _uiState.update { old -> old.copy(data = settings) }
+                }
             }
         }
     }
 
+    /**
+     * Updates a specific setting with a new value.
+     *
+     * @param title The title or key of the setting to be updated.
+     * @param newValue The new value for the specified setting.
+     */
     fun writeNewSetting(title: String, newValue: Any) {
         viewModelScope.launch {
             repo.addSetting(title, newValue)
         }
     }
 
+    /**
+     * Resets all user settings to their default values.
+     */
     fun resetSettings() {
         viewModelScope.launch {
             repo.resetSettings()
         }
     }
 
+    /**
+     * Logs out the current user, cancels ongoing workouts, and triggers UI navigation back.
+     */
     fun logout() {
         viewModelScope.launch {
             auth.logout()
             workoutState.cancel()
-            _state.postValue(State.Navigate(R.id.settings_to_welcome))
+            _uiState.update { old -> old.copy(returnBack = true) }
```

Comment:
you can trigger navigateBack directly from composable login callback

Reply:
there is no longer a return back state